apps/arweave/src/ar_mining_server.erl

```diff
@@ -391,56 +409,38 @@ update_cache_limits(State) ->
 update_cache_limits(0, State) ->
 	State;
 update_cache_limits(NumActivePartitions, State) ->
-	%% This allows the cache to store enough chunks for 4 concurrent VDF steps per partition.
-	IdealStepsPerPartition = 4,
-	IdealRangesPerStep = 2,
-	%% The number of the sub-chunks in the recall range for packing difficulty = 0 is
-	%% the largest. Each mining worker will reduce it according to its packing difficulty
-	%% in set_cache_limits.
-	ChunksPerRange = ar_mining_worker:recall_range_sub_chunks(0),
-	IdealCacheLimit = ar_util:ceil_int(
-		IdealStepsPerPartition * IdealRangesPerStep * ChunksPerRange * NumActivePartitions, 100),
-
-	{ok, Config} = application:get_env(arweave, config),
-	OverallCacheLimit = case Config#config.mining_server_chunk_cache_size_limit of
-		undefined ->
-			IdealCacheLimit;
-		N ->
-			N
-	end,
-
-	%% We shard the chunk cache across every active worker. Only workers that mine a partition
-	%% included in the current weave are active.
-	NewCacheLimit = max(1, OverallCacheLimit div NumActivePartitions),
+	{MinimumCacheLimitMiB, OverallCacheLimitMiB, PartitionCacheLimit, VDFQueueLimit,
+		GarbageCollectionFrequency} =
+			calculate_cache_limits(NumActivePartitions,  State#state.packing_difficulty),
 
-	case NewCacheLimit == State#state.chunk_cache_limit of
+	case PartitionCacheLimit == State#state.chunk_cache_limit of
```

Comment:
These parts of the code are easily extracted into sharp functions like
```erlang
maybe_update_cache_limit(PartitionCacheLimit, #state{chunk_cache_limit = PartitionCacheLimit} = S0) -> S0;
maybe_update_cache_limit(PartitionCacheLimit, S0) ->
    ...
```
This will greatly reduce the nesting within a function and reduce function scope.

Reply:
Done! https://github.com/ArweaveTeam/arweave/pull/636/commits/1a8a2ce25b993da9e0cec6282ed00abbf28da073